src/core_modules/capture-core/components/DataEntries/EnrollmentRegistrationEntry/hooks/useFormValues.js

```diff
@@ -128,8 +128,8 @@ const buildFormValues = async ({
         staticPatternValues,
         querySingleResource,
     );
-    setFormValues && setFormValues({ ...formValues, ...uniqueValues, ...searchFormValues });
-    setClientValues && setClientValues({ ...clientValues, ...uniqueValues, ...searchClientValues });
+    setFormValues && setFormValues({ ...searchFormValues, ...formValues, ...uniqueValues });
```

Comment:
Hey,
Any reason why you changed the order of these objects? Because `uniqueValues` is now the rightmost object, it will overwrite previous existing unique IDs. From what I remember, if there are unique IDs inside `searchFormValues`, we want to reuse them.
Thanks!

Reply:
Oops, I didn't properly read your explanation in the description 😅 . It makes sense now. The possible scenarios I am aware of are:
- enroll a new TEI
- re-enroll an existing TEI
- search and enroll a new TEI
- search and re-enroll an existing TEI

I tested all of these and your bug fix looks good to me. Thanks!